### ISSUE TITLE
Bug with mysql on recent ubuntu servers

### ISSUE THREAD

**Issue by purplefish32** [claroline/Distribution#424]
When installing Claroline Connect on Ubuntu 16.04 that has a recent version of mysql , you get the following error if you try to open the resources page.

```
Expression #16 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'claroline.c5_.date_log' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by
```

This can be fixed by changing the mysql conf but it seems better to me to fix this error on a doctrine/claro level.

**Comment by purplefish32** [comment]
See : http://stackoverflow.com/questions/34115174/i-am-getting-an-error-in-mysql-related-to-only-full-group-by-when-executing-a-qu

**Comment by Elorfin** [comment]
You need to change you mysql config to be able to use aliases in group by.

(It seems complicated to not use aliases here because for most queries it's Doctrine which generates them).

@pitrackster have get this same error.

**Comment by stefk** [comment]
Here's a good explanation: http://www.tocker.ca/2014/01/24/proposal-to-enable-sql-mode-only-full-group-by-by-default.html

TLDR: pre-5.7 behaviour is "non-standard", "non-deterministic" and "can hide bugs in application code".

So +1 for dealing with that in our code.

Other resource: https://www.drupal.org/node/567148

**Comment by ngodfraind** [comment]
The error is self explanatory.

Upgrade to doctrine/orm and octrine/dbal latest version didn't fix anything.

I don't really know how to fix it properly (and my guess is that we should run database query tests while fixing it).

This is usually caused by a GROUP BY clause when we do a plugin activation check.
For instance, in the resource manager, when we open a directory it crashes because 

```
 (CONCAT(c7_.vendor_name, c7_.short_name) IN (?, ?, ?, ?, ?, ?)
```

obviously requires an array and there is a GROUP BY later.
Simply removing the GROUP BY fixes the error but I don't think it's a long term solution. 

https://github.com/claroline/Distribution/blob/master/main/core/Repository/ResourceNodeRepository.php#L182

**Comment by stefk** [comment]
This isn't related to the plugin part IMO. The problem occurs when columns in the SELECT clause aren't used with aggregating functions *and* aren't present in the GROUP BY clause. The actual error message mentions the `date_log` column, but even if it was fixed for that column, every other one in the SELECT clause will probably produce the same error.

> Simply removing the GROUP BY fixes the error but I don't think it's a long term solution.

In this case, that seems logical. Why having a GROUP BY if there's no aggregating function in the SELECT list?

**Comment by LaurentGruber** [comment]
a workaround is to set mysql in previous mode 
`set global sql_mode='' `
